src/Avalonia.Controls/Window.cs

```diff
@@ -365,19 +369,59 @@ internal void Close(bool ignoreCancel)
             {
                 if (close)
                 {
-                    PlatformImpl?.Dispose();
+                    CloseInternal();   
                 }
             }
         }
 
+        private void CloseInternal ()
+        {
+            foreach(var child in _children)
```

Comment:
Make a copy of the collection since child will call `owner.RemoveChild` and we might get `InvalidOperationException` for "collection was modified"

Reply:
fixed